plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/LeasedPlcConnection.java

```diff
@@ -92,15 +102,139 @@ public PlcReadRequest.Builder readRequestBuilder() {
         if(connection == null) {
             throw new PlcRuntimeException("Error using leased connection after returning it to the cache.");
         }
-        return connection.readRequestBuilder();
+        final PlcReadRequest.Builder innerBuilder = connection.readRequestBuilder();
```

Comment:
Could you please explain what this is needed for?

Reply:
This is for handling  request failure and close the raw connection